File: workspace/2023-01-03/fizz-buzz.forth/fizz-buzz.fs

: fizz-buzz-value ( n -- x )
    dup 15 mod 0 = if
        drop
        s" FizzBuzz " type
    else
        dup 5 mod 0 = if
            drop
            s" Buzz " type
        else
            dup 3 mod 0 = if
                drop
                s" Fizz " type
            else
                .
            endif
        endif
    endif
;

: fizz-buzz-range recursive ( min max -- x_min ... x_max )
    over over <= if
        over fizz-buzz-value    ( min max -- min max x )
        swap 1 + swap           ( min max -- min+1 max )
        fizz-buzz-range
    else
        drop drop               ( min max -- )
    endif
;

: fizz-buzz ( n -- x_1 x_2 ... x_n )
    1 swap fizz-buzz-range
;

100 fizz-buzz

File: workspace/2023/2023-05-07/numerical-sequence.prolog/fizzbuzz.pl

fizzbuzz(N, Fn) :-
    (
        N mod 15 =:= 0 ->
            Fn = "FizzBuzz";
        N mod 5 =:= 0 ->
            Fn = "Buzz";
        N mod 3 =:= 0 ->
            Fn = "Fizz";
        otherwise ->
            number_string(N, Fn)
    ).

fizzbuzz_values(N, Xs) :-
    fizzbuzz_values(1, N, Xs).

fizzbuzz_values(Min, Max, []) :-
    Min > Max.

fizzbuzz_values(Min, Max, [X|Xs]) :-
    Min =< Max,
    fizzbuzz(Min, X),
    is(NextMin, Min + 1),
    fizzbuzz_values(NextMin, Max, Xs).

